Test ivy 58

masonlalonde17 opened the pull request:
Creates test folder to hold all tests. IVY-58 test created.

masonlalonde17 committed:
First commit

masonlalonde17 committed:
First commit

masonlalonde17 committed:
tests commit
Created a test folder that holds all tests. Tests are named after the branch they are created for

masonlalonde17 committed:
tests commit
Created a test folder that holds all tests. Tests are named after the branch they are created for

mmacdo52 committed:
Update schema.prisma

Added the password column to the user table

camfiander commented on tests/IVY-58.spec.js:
```diff
@@ -0,0 +1,39 @@
+const { threadId } = require('worker_threads');
+const { SeleniumServer } = require('selenium-webdriver/remote');
+
+suite(function(env) {
```
You may also have to move the tests folder into `client` so that mocha uses dependencies defined in `client/package.json`

camfiander reviewed (changes requested):
It took me a bit of setup before I got the tests to run and pass, but this is a great start!

There are a couple files that shouldn't have been modified here:
- `.DS_Store` is a macOS artifact, can you remove this and add a line to the `.gitignore`, so that it gets ignored for future commits?
- `server/package-lock.json` shouldn't be modified unless there were changes to `server/package.json`. If I pull in your branch and run `cd server` then `npm install`, the `package-lock.json` changes. You should be able to revert these changes like that.

We should do a quick call to make sure these tests are set up to run in CI as well

Commit:
Merge branch 'main' into data-model

Conflicts:
	server/prisma/schema.prisma

Commit:
add oncall

masonlalonde17 committed:
moved tests into client folder

masonlalonde17 committed:
Merge branch 'main' of https://github.com/ivysaur-unb/on-call-coordinator into test-IVY-58

masonlalonde17 committed:
tests folder renamed to test

masonlalonde17 committed:
added mocha to dependencies

masonlalonde17 committed:
added selenium-webdriver and chromedriver

masonlalonde17 committed:
deleted extra env file

masonlalonde17 committed:
test IVY-58 changes

masonlalonde17 committed:
test-IVY-58 changes

masonlalonde17 committed:
test-IVY-58 changes

masonlalonde17 committed:
tets folder outside of client deleted

Conner-H committed:
Updated lots of things | heavilly messed with how other things work.

-Made new font & applied to everything
- Standardization across the board
- choose working colour combos to work with colour blindness
-reformatted a fair bit of css and and combined a bunch of peoples CSS

Conner-H committed:
pushing the commit issues here

.

Conner-H committed:
Revert "pushing the commit issues here"

This reverts commit 32ebc77425acbeeac998400e1b1c75f870c22a73.

camfiander committed:
Add login test

Conner-H committed:
fixed and checked all commit errors that existed prviously

Conner-H committed:
Fonts broke again, fixed them & what broke in scheduleAbsences

Conner-H committed:
changed colours of dashboard based on group recommendation

camfiander committed:
ignore eslint

gracekiely committed:
Dropdown with teachers displayed working

gracekiely committed:
displaying on calls not working

gracekiely committed:
package

gracekiely committed:
removed server import

gracekiely committed:
Started page for story 33 - teacher view on calls

masonlalonde17 committed:
Merge pull request #34 from ivysaur-unb/test-IVY-58-Cam

Add login test

gracekiely committed:
get request for on calls working so far

gracekiely committed:
fixed UploadClasses read-excel-file

Conner-H committed:
set up theme stuff for new header

gracekiely committed:
day of on call is displaying

gracekiely committed:
scheduledClassId and day displaying

gracekiely committed:
displays logged in teachers oncalls

gracekiely committed:
correct teacher selected when button clicked

gracekiely committed:
removed comments

gracekiely committed:
displaying period and location for oncalls and ui

masonlalonde17 committed:
IVY-58 test updates
Added some chagnes to the code to now pass the initial tests

masonlalonde17 committed:
Merge branch 'main' of https://github.com/ivysaur-unb/on-call-coordinator into test-IVY-58

gracekiely committed:
css changes

Conner-H committed:
Merge branch 'main' into IVYSAUR-109

Conner-H committed:
fixing merge conflict erros

camfiander committed:
Merge remote-tracking branch 'origin/main' into IVYSAUR-7-VP-view-on-calls

MitchellCWM committed:
merged main

MitchellCWM committed:
fixed problem with teachers not displayed their onCalls. The problem was that is was using userId and not teacherId

MitchellCWM committed:
removed a comments

MitchellCWM committed:
removed more comments

MitchellCWM committed:
removed another comment and console log

MitchellCWM committed:
deleted prisma migrations folder, sorry

MitchellCWM committed:
prisma migrate

MitchellCWM committed:
merged main

MitchellCWM committed:
stlyed the create teacher text fields

MitchellCWM committed:
create school is styled

MitchellCWM committed:
absence table buttons are back to the correct colors

MitchellCWM committed:
fixed some icon colors. Fixed a bug when selecting the teachables for creating a teacher there were two scroll bars

MitchellCWM committed:
fixed color on edit icon for absence schedule and updated the color for the log out on the header

MitchellCWM committed:
updated the styling for the create teacher profile

MitchellCWM committed:
updated the styling of the on call table for the VP

MitchellCWM committed:
started styling the vp oncall. Also fixed a bug where if schedule class had no class, then everything would break. Now it just prints Untitled

MitchellCWM committed:
updated styling for the view oncall for vice principal

MitchellCWM committed:
updated the secondary color - the primary color is white and the secondary is the blue

MitchellCWM committed:
fixed a bug where when creating a teacher, if they did not have schoolId, the schoolId would be 1 even if there was no school with an id of 1

MitchellCWM committed:
the schedule will now work on the GUI

camfiander committed:
Fix migrations

MitchellCWM committed:
neatened oncall

MitchellCWM committed:
for viewing on-calls if the schedule does not have a course code, it will display the special code

MitchellCWM committed:
Merge branch 'IVYSAUR-109' of https://github.com/ivysaur-unb/on-call-coordinator into IVYSAUR-109

camfiander committed:
Use one instance of prisma

Too many connections breaking unit tests

camfiander committed:
oh the wonders of async/await

plz don't try making 30 prisma calls simultaneously

camfiander committed:
make view oncalls reactive

masonlalonde17 committed:
test changes

camfiander committed:
er diagram

masonlalonde17 committed:
test changes again

masonlalonde17 committed:
Merge branch 'test-IVY-58' of https://github.com/ivysaur-unb/on-call-coordinator into test-IVY-58

masonlalonde17 committed:
Merge pull request #43 from ivysaur-unb/IVYSAUR-109

Ivysaur 109

masonlalonde17 committed:
Working test changes

masonlalonde17 committed:
Merge branch 'test-IVY-58' of https://github.com/ivysaur-unb/on-call-coordinator into test-IVY-58

masonlalonde17 committed:
ER diagram changes

masonlalonde17 committed:
It finally worked

masonlalonde17 committed:
trying CI test again
